js/client/modules/@arangodb/testutils/testrunner.js

```diff
@@ -397,44 +75,39 @@ class testRunner {
     this.env = {};
     this.results = {};
     this.continueTesting = true;
-    this.usersCount = 0;
-    this.taskCount = 0;
-    this.cleanupChecks = [ ];
-    this.cleanupChecks.push(failurePointsCheck);
-
-    if (checkCollections) {
-      this.cleanupChecks.push(tasksTests);
-    }
-    if (checkUsers) {
-      this.cleanupChecks.push(usersTests);
-    }
-    if (this.options.agency) {
-      // pure agency tests won't need user checks.
-      if (this.options.cluster) {
-        this.cleanupChecks.push(databasesTest);
+    this.instanceManager = undefined;
+    this.cleanupChecks = this.loadSutChecks(disableChecks);
+  }
+  loadSutChecks(disableCheckFilter) {
+    let sutCheckers = _.filter(fs.list(fs.join(__dirname, 'sutcheckers')),
+                              function (p) {
+                                let extension = p.substr(-3);
+                                let basename = p.slice(0, -3);
+                                return (
+                                  (extension === '.js') &&
+                                  (disableCheckFilter.filter(f => f === basename).length === 0)
+                                );
+                              }).sort();
+    let ret = [];
+    sutCheckers.forEach(fn => {
+      try {
+        let checker = require('@arangodb/testutils/sutcheckers/'+fn).checker;
+        ret.push(new checker(this));
+      } catch (x) {
+        print('failed to load module ' + fn);
+        throw x;
       }
-    } else {
-      this.cleanupChecks.push(databasesTest);
-    }
-    this.collectionsBefore = [];
-    if (checkCollections) {
-      this.cleanupChecks.push(collectionsTest);
-    }
-    this.viewsBefore = [];
-    if (checkCollections) {
-      this.cleanupChecks.push(viewsTest);
-    }
-    this.analyzersBefore = [];
-    if (checkCollections) {
-      this.cleanupChecks.push(analyzersTest);
-    }
-    this.graphCount = 0;
-    if (checkCollections) {
-      this.cleanupChecks.push(graphsTest);
+    });
+    return ret;
+  }
+  setResult(te, serverDead, res) {
+    let orgRes = JSON.stringify(this.results[this.translateResult(te)]);
+    this.results[this.translateResult(te)] = res;
+    if (serverDead) {
+      this.serverDead = true;
     }
-    this.instanceManager = undefined;
+    this.results[this.translateResult(te)].message += " ] - Original test status: \n" + orgRes;
```

Comment:
These extra brackets in various places seem just weird. We should remove them alltogether
```suggestion
    this.results[this.translateResult(te)].message += " - Original test status: \n" + orgRes;
```